server/src/domain/user/user.service.ts

```diff
@@ -93,10 +93,25 @@ export class UserService {
     authUser: AuthUserDto,
     fileInfo: Express.Multer.File,
   ): Promise<CreateProfileImageResponseDto> {
+    const { profileImagePath: oldpath } = await this.findOrFail(authUser.id, { withDeleted: false });
     const updatedUser = await this.userRepository.update(authUser.id, { profileImagePath: fileInfo.path });
+    if (oldpath !== '') {
+      const files = [oldpath];
+      await this.jobRepository.queue({ name: JobName.DELETE_FILES, data: { files } });
+    }
     return mapCreateProfileImageResponse(updatedUser.id, updatedUser.profileImagePath);
   }
 
+  async deleteProfileImage(authUser: AuthUserDto): Promise<void> {
+    const user = await this.findOrFail(authUser.id, { withDeleted: false });
+    if (user.profileImagePath !== '') {
```

Comment:
I think an http error make be more appropriate if the user doesn't have a profile picture. The client can ignore it instead.